src/test/kotlin/com/makarytskyi/rentcar/repository/OrderRepositoryTest.kt

```diff
@@ -27,141 +27,170 @@ internal class OrderRepositoryTest : ContainerBase {
+        val car2 = carRepository.create(randomCar()).block()
+        val user2 = userRepository.create(randomUser()).block()
+        val order2 = orderRepository.create(randomOrder(car2?.id, user2?.id)).block()
 
         // WHEN
-        val orders = orderRepository.findAll(0, 20)
+        val allOrders = orderRepository.findFullAll(0, 20)
 
         // THEN
-        assertTrue(orders.any { it.id == order1.id })
-        assertTrue(orders.any { it.id == order2.id })
+        StepVerifier.create(allOrders.collectList())
+            .assertNext { orders ->
+                assertTrue(orders.any { it.id == order1?.id })
+                assertTrue(orders.any { it.id == order2?.id })
+            }
+            .verifyComplete()
     }
 
     @Test
     fun `patch should partially update order`() {
         // GIVEN
         val from = monthAfter
         val to = monthAndDayAfter
-        val car = carRepository.create(randomCar())
-        val user = userRepository.create(randomUser())
-        val order = orderRepository.create(randomOrder(car.id, user.id))
+        val car = carRepository.create(randomCar()).block()
+        val user = userRepository.create(randomUser()).block()
+        val order = orderRepository.create(randomOrder(car?.id, user?.id)).block()
 
         val updateOrder = emptyOrderPatch().copy(
             from = from,
             to = to
         )
 
         // WHEN
-        val updated = orderRepository.patch(order.id.toString(), updateOrder)
+        val updated = orderRepository.patch(order?.id.toString(), updateOrder)
 
         // THEN
-        assertEquals(from, updated?.from)
-        assertEquals(to, updated?.to)
+        StepVerifier.create(updated)
+            .assertNext {
+                assertEquals(from, it.from)
+                assertEquals(to, it.to)
+            }
+            .verifyComplete()
     }
 
     @Test
     fun `findById should return existing order by id`() {
         // GIVEN
-        val car = carRepository.create(randomCar())
-        val user = userRepository.create(randomUser())
-        val order = orderRepository.create(randomOrder(car.id, user.id))
+        val car = carRepository.create(randomCar()).block()
+        val user = userRepository.create(randomUser()).block()
+        val order = orderRepository.create(randomOrder(car?.id, user?.id)).block()
 
         // WHEN
-        val foundOrder = orderRepository.findById(order.id.toString())
+        val foundOrder = orderRepository.findFullById(order?.id.toString())
 
         // THEN
-        assertEquals(order.id, foundOrder?.id)
-        assertEquals(order.carId, foundOrder?.car?.id)
-        assertEquals(order.userId, foundOrder?.user?.id)
+        StepVerifier.create(foundOrder)
+            .assertNext {
+                assertEquals(order?.id, it.id)
+                assertEquals(order?.carId, it.car?.id)
+                assertEquals(order?.userId, it.user?.id)
```

Comment:
Why not comparing full objects?

Reply:
They have different types: MongoOrder and AggregatedMongoOrder.